src/Form/EntityEmbedDialog.php

```diff
@@ -306,13 +321,27 @@ public function validateForm(array &$form, FormStateInterface $form_state) {
             if (!$this->accessEntity($entity, 'view')) {
               $form_state->setError($form['attributes']['data-entity-id'], $this->t('Unable to access @type entity @id.', array('@type' => $entity_type, '@id' => $id)));
             }
-            elseif ($uuid = $entity->uuid()) {
-              $form_state->setValueForElement($form['attributes']['data-entity-uuid'], $uuid);
-              $form_state->setValueForElement($form['attributes']['data-entity-id'], $entity->id());
-            }
             else {
-              $form_state->setValueForElement($form['attributes']['data-entity-uuid'], '');
               $form_state->setValueForElement($form['attributes']['data-entity-id'], $entity->id());
+              if ($uuid = $entity->uuid()) {
+                $form_state->setValueForElement($form['attributes']['data-entity-uuid'], $uuid);
+              }
+              else {
+                $form_state->setValueForElement($form['attributes']['data-entity-uuid'], '');
+              }
+
+              // Ensure that at least one display plugin is present before
+              // proceeding to the next step. Rasie an error otherwise.
+              $embed_button = $form_state->get('embed_button');
+              $allowed_plugins = $embed_button->getAllowedDisplayPlugins();
+              $available_plugins = $this->displayPluginManager()->getDefinitionOptionsForEntity($entity);
+              $display_plugin_options = empty($allowed_plugins) ? $available_plugins : array_intersect_key($available_plugins, $allowed_plugins);
+              // If no plugin is available after taking the intersection,
+              // raise error. Also log an exception.
+              if (empty($display_plugin_options)) {
+                $form_state->setError($form['attributes']['data-entity-id'], $this->t('No display options available for the selected entity. Please select another entity.'));
+                $this->logger->warning('No display options available for "@type:" entity "@id" while embedding using button "@button". Please ensure that at least one display plugin is allowed for this embed button which is available for this entity.', array('@type' => $entity_type, '@id' => $entity->id(), '@button' => $embed_button->id()));
```

Comment:
Thats not required per se but just added them so that the user can distinguish between the actual values and rest of the text.